lib/api/unittest/CAnomalyJobLimitTest.cc

```diff
@@ -344,147 +344,173 @@ void CAnomalyJobLimitTest::testModelledEntityCountForFixedMemoryLimit() {
     // create for a small(ish) memory limit to catch large changes in
     // the memory used per partition of the data.
 
-    core_t::TTime bucketLength{600};
-    TGeneratorVec generators{periodic, tradingDays, level, ramp, sparse};
-    std::stringstream outputStrm;
-    core::CJsonOutputStreamWrapper wrappedOutputStream(outputStrm);
-    api::CAnomalyJob::TStrStrUMap dataRows;
-    TSizeVec generator;
+    struct STestParams {
+        core_t::TTime s_BucketLength;
+        std::size_t s_ExpectedByFields;
+        std::size_t s_ExpectedOverFields;
+        std::size_t s_ExpectedPartitionFields;
+        std::size_t s_ExpectedByLowerMemoryLimit;
+        std::size_t s_ExpectedPartitionLowerMemoryLimit;
+        std::size_t s_ExpectedOverLowerMemoryLimit;
+    } testParams[] = {{600, 550, 6000, 300, 33, 40, 40},
+                      {3600, 550, 5800, 300, 30, 25, 20},
+                      {7200, 550, 5000, 300, 30, 26, 10},
+                      {172800, 150, 850, 120, 6, 6, 3},
+                      {604800, 150, 850, 120, 6, 6, 3}};
```

Comment:
This is great, but I wonder how long this unit test runs for now? This was already pretty long, it may be we want to cut the cases, or only optionally run some of them, if the total runtime is very high.

Reply:
I've trimmed a few of the cases - runtime is now ~35s (was originally ~30s).